aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -96,6 +98,38 @@ def define(cls, spec):
             help='The successfully relaxed structure, unless `relax_type is RelaxType.NONE`.')
         # yapf: enable
 
+    @classmethod
+    def get_builder_from_protocol(cls, code, structure, protocol=None, overrides=None, **kwargs):
+        """Return a builder prepopulated with inputs selected according to the chosen protocol.
+
+        :param code: the ``Code`` instance configured for the ``quantumespresso.pw`` plugin.
+        :param structure: the ``StructureData`` instance to use.
+        :param protocol: protocol to use, if not specified, the default will be used.
+        :param overrides: optional dictionary of inputs to override the defaults of the protocol.
+        :param kwargs: additional keyword arguments that will be passed to the ``get_builder_from_protocol`` of all the
+            sub processes that are called by this workchain.
+        :return: an process builder instance with all inputs defined ready for launch.
+        """
+        args = (code, structure, protocol)
+        inputs = cls.get_protocol_inputs(protocol, overrides)
+        builder = cls.get_builder()
+
+        base = PwBaseWorkChain.get_builder_from_protocol(*args, overrides=inputs.get('base', None), **kwargs)
```

Comment:
So, we don't allow separate overrides for the `PwBaseWorkChain`s of the geometry optimization and the `base_final_scf`?

As a side note, now that I've been working on the `create_magnetic_allotrope` calcfunction some more, I am wondering if we have to pass the magnetization between the various work chains inside the `PwRelaxWorkChain`. There are two questions I have here:

1. Does resetting the `starting_magnetization` have a large influence on the final magnetization? Do we have to create new magnetic kinds here as well? This can be checked by running a separate final scf `PwBaseWorkChain` that starts from the structure passed to the `final_scf` work chain but uses the `output_trajectory` from the final iteration of the geometry optimization work chains.
2. Does the magnetization have an influence on the geometry? I.e. in case the magnetization is different, what are the final forces we find for the final scf?

Frankly, I don't think the differences will be dramatic, but it's probably a good idea to check. But that's definitely beyond the scope of this PR. 😅

Reply:
Good point, I simply forgot about that one. I will add it because I think it would be good to run that by default. The user can always get rid of it by popping those inputs from the builder returned, just as in the case of the bands workchain where you can remove the relax namespace to not perform geometry optimization.

Regarding the side note: I think you may be right that the relax (or the base workchain for that matter) does not update the starting magnetization. So without checking the code, I think it may actually use the original starting magnetization passed to the top level workflow at every step, essentially ignoring what was computed. It would indeed be good to verify this and fix it if this is the case. However, I am afraid that this might complicate the code considerable since we will effectively have to call the `requires_new_magnetic_kinds` and `create_magnetic_allotrope` within the workchain :S

Reply:
Well, one solution would be to have the magnetic configuration be stored the `StructureData` class. 🙃 
Maybe we should open an `aiida-core` discussion on this, i.e. which structural properties should be a part of the `StructureData`?